streams/src/main/java/org/apache/kafka/streams/kstream/internals/GroupedStreamAggregateBuilder.java

```diff
@@ -76,13 +62,14 @@ public V apply() {
         this.streamsGraphNode = streamsGraphNode;
     }
 
-    <T> KTable<K, T> build(final KStreamAggProcessorSupplier<K, ?, V, T> aggregateSupplier,
-                           final String functionName,
-                           final StoreBuilder storeBuilder,
-                           final boolean isQueryable) {
+    <KR, T> KTable<KR, T> build(final KStreamAggProcessorSupplier<K, KR, V, T> aggregateSupplier,
```

Comment:
We previously discarded the knowledge of what the result key type is. It turns out that we needed it, and we were just casting the result to re-introduce it.